### ISSUE TITLE
take_over_control doesn't work anymore

### ISSUE THREAD

**Issue by FPF92** [basnijholt/adaptive-lighting#377]
#### Version information:
1.2.1

#### Description:
When i change the brightness and / or color temperature of my lights that are setup in adaptive light these parameters fallback after seconds. 

I turned on the option:
take_over_control: If anything but Adaptive Lighting calls 'light.turn_on' when a light is already on, stop adapting that light until it (or the switch) toggles off -> on.

I assume it started after the update to version 1.2.x because in the past it worked fine.

**Comment by msl2424** [comment]
I have the same issue, at least with brightness, in addition to lights now randomly turning on. The inability to use take_over_control has persisted for as long as I can remember, whereas lights turning on randomly is a new issue. I have Philips Hue lights connected to a Hue hub, and I manually control them using a Lutron Aurora Smart Bulb Dimmer Switch (for adjusting the brightness, and turning on and off). Take_over_control and detect_non_ha_changes have always been enabled. Really hoping there is a solution, especially for take_over_control for manual brightness changes.

**Comment by Gerben321** [comment]
Same here. Also random lights turning back on. New install for two days now. If it keeps happening I'll remove it for a while until it's fixed. Great potential!

**Comment by robvangeloven** [comment]
The lights turning on randomly seems to be linked to an adaptive lighting timer not being reset / stopped after manually turning off a device, at least for me.

**Comment by djcrawleravp** [comment]
I can't turn off my lights, after a couple of seconds It turns on again

**Comment by mlangenberg** [comment]
Just switched from [hass-circadian_lighting](https://github.com/claytonjn/hass-circadian_lighting) to adaptive-lighting, because I needed something that would automatically override the adaptive lighting when manually adjusting the brightness from the home assistant user interface. 

I can confirm that enabling `take_over_control` does not do anything at the moment using version `v1.4.1`.

**Comment by ChipWolf** [comment]
Also confirming, not seeing any `adaptive_control.manual_control` events either, but the state changed events are coming through from the non_ha changes

**Comment by brebtatv** [comment]
I have identified the problem.
If the "interval" setting is longer (seconds) than the "transition" or "sleep_transition" duration, the manual_control switch will not have time to set to true or to the given lights. The manual mode is thus not activated.

But if we set the "interval" to a time shorter than "transition" and "sleep_transition", then manual_control has time to write and the manual mode is already functional.

**Comment by djcrawleravp** [comment]
> I have identified the problem. If the "interval" setting is longer (seconds) than the "transition" or "sleep_transition" duration, the manual_control switch will not have time to set to true or to the given lights. The manual mode is thus not activated.
> 
> But if we set the "interval" to a time shorter than "transition" and "sleep_transition", then manual_control has time to write and the manual mode is already functional.

Sorry can you post a screenshot of the parameters you have changed?

THanks in advance!

**Comment by basnijholt** [comment]
> If the "interval" setting is longer (seconds) than the "transition" or "sleep_transition" duration, the manual_control switch will not have time to set to true or to the given lights. The manual mode is thus not activated.

This should be easily fixable in the code too. Just need to find some time ;)

**Comment by basnijholt** [comment]
I made the appropriate change here: https://github.com/basnijholt/adaptive-lighting/pull/417

If someone could test it, it would be great.

**Comment by djcrawleravp** [comment]
Hello guys, I've tested the lights again but the problem still persist.

The light and lightgroups are turning on after a couple of seconds of turningh them off manually

**Comment by Steini1984** [comment]
> Hello guys, I've tested the lights again but the problem still persist.
> 
> The light and lightgroups are turning on after a couple of seconds of turningh them off manually
I can confirm that this is also happening to me. Using Zigbee2mqtt with ikea Tradfri lights in a zigbee group and (at least) this is happening when turning the lights off manually via a wall switch that is directly bound to the lights. 

Here are my settings:
![image](https://user-images.githubusercontent.com/5666926/213126672-00dbf2e7-3f45-44b1-b48d-7b5f49c24927.png)

**Comment by DaveNankervis** [comment]
> I have identified the problem. If the "interval" setting is longer (seconds) than the "transition" or "sleep_transition" duration, the manual_control switch will not have time to set to true or to the given lights. The manual mode is thus not activated.
> 
> But if we set the "interval" to a time shorter than "transition" and "sleep_transition", then manual_control has time to write and the manual mode is already functional.

After updating the full example to have transition: 120 leaving the interval: 90 this seems to have fixed this issue for me.

**Comment by th3w1zard1** [comment]
@basnijholt Hey! I tried your fix but it didn't work. Here's the logs!

```
Logger: homeassistant.components.switch
Source: custom_components/adaptive_lighting/switch.py:680
Integration: Switch ([documentation](https://www.home-assistant.io/integrations/switch), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+switch%22))
First occurred: 2:03:41 AM (6 occurrences)
Last logged: 2:03:41 AM

Error while setting up adaptive_lighting platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 293, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/adaptive_lighting/switch.py", line 410, in async_setup_entry
    switch = AdaptiveSwitch(
  File "/config/custom_components/adaptive_lighting/switch.py", line 680, in __init__
    if self._interval > self._transition or self._interval > self._sleep_transition:
AttributeError: 'AdaptiveSwitch' object has no attribute '_transition'
```

**Comment by th3w1zard1** [comment]
Fixed in version [1.10.0](https://github.com/basnijholt/adaptive-lighting/releases/tag/1.10.0)
